petastorm/tests/test_spark_dataset_converter.py

```diff
@@ -24,11 +24,15 @@ def test_primitive(self):
             StructField("double_col", DoubleType(), False),
             StructField("short_col", ShortType(), False),
             StructField("int_col", IntegerType(), False),
-            StructField("long_col", LongType(), False)
+            StructField("long_col", LongType(), False),
+            StructField("str_col", StringType(), False),
+            StructField("bin_col", BinaryType(), False),
+            StructField("byte_col", ByteType(), False),
         ])
         df = self.spark.createDataFrame([
-            (True, 0.12, 432.1, 5, 5, 0),
-            (False, 123.45, 0.987, 9, 908, 765)], schema=schema).coalesce(1)
+            (True, 0.12, 432.1, 5, 5, 0, "hello", bytearray(b"spark"), -128),
```

Comment:
You can use `b"spark"` instead of `bytearray(b"spark")`.

Reply:
We'd better test some special characters like 
`b"spark\x01\x02spark"`

Reply:
Using b'spark' doesn't work in CI tests... but it works locally.